graql/reasoner/atom/Atom.java

```diff
@@ -70,20 +70,20 @@
     private final AtomValidator<Atom> validator = new BasicAtomValidator();
     private Set<InferenceRule> applicableRules = null;
 
-    private final ConceptId typeId;
+    private final Label typeLabel;
 
-    public Atom(ReasonerQuery reasonerQuery, Variable varName, Statement pattern, ConceptId typeId, ReasoningContext ctx) {
+    public Atom(ReasonerQuery reasonerQuery, Variable varName, Statement pattern, @Nullable Label typeLabel, ReasoningContext ctx) {
```

Comment:
why is this nullable?

Reply:
`$x isa $type;` :)

Reply:
Basically if we do not know the non-meta type of the atom.